3_ecosystem/3_2_macro/src/main.rs

```diff
@@ -1,3 +1,25 @@
-fn main() {
-    println!("Implement me!");
-}
+macro_rules! btreemap {
+    ($( $key:expr => $val:expr ),*) => {
+        {
+            let mut btreemap = std::collections::BTreeMap::new();
+            $(
+                btreemap.insert($key,$val);
+            )*
+            btreemap
+        }
+    };
+}
+
+fn main() {
+    let map1 = btreemap!(
+        "k1" => "v1",
+        "k2" => "v2"
+    );
```

Comment:
Фигурные скобки в таких макросах более привычный вариант:

```suggestion
    let map1 = btreemap! {
        "k1" => "v1",
        "k2" => "v2"
    };
```